tsconfig.json

```diff
@@ -1,10 +1,12 @@
-// Based on node 16 config: https://github.com/tsconfig/bases/blob/main/bases/node16.json
+// Based on node 18 config: https://github.com/tsconfig/bases/blob/main/bases/node18.json
 {
     "$schema": "https://json.schemastore.org/tsconfig",
-    "display": "Node 16",
+    "display": "Node 18",
     "compilerOptions": {
-        "target": "es2021",
-        "module": "NodeNext",
+        "lib": ["es2023"],
+        "module": "node16",
+        "moduleResolution": "node16",
+        "target": "es2022",
```

Comment:
Changing from 2021 to 2022 could have an impact on the minimum version of node that we support, basically changing the minor version of node 16 that is required.  Whether we actually _use_ any of the features that aren't supported in early 16 versions, I'm not sure.  So maybe best to save this for a breaking version later on.